src/vsr/replica.zig

```diff
@@ -5597,6 +5612,12 @@ pub fn ReplicaType(
                 assert(self.journal.has(header));
             }
 
+            if (self.checkpoint_id_for_op(header.op)) |checkpoint_id| {
+                assert(header.checkpoint_id == checkpoint_id);
+            } else {
+                assert(header.op < self.op_repair_min());
```

Comment:
I don't get this assert in the `else` branch, I think it's tautologically false here? 

```
            if (header.op < self.op_repair_min()) {
                // Slots too far back belong to the next wrap of the log.
                log.debug("{}: repair_header: false (precedes op_repair_min={})", .{
                    self.replica,
                    self.op_repair_min(),
                });
                return false;
            }

            if (self.checkpoint_id_for_op(header.op)) |checkpoint_id| {
                assert(header.checkpoint_id == checkpoint_id);
            } else {
                assert(header.op < self.op_repair_min());
            }
```